distreqx/distributions/categorical.py

```diff
@@ -0,0 +1,213 @@
+"""Categorical distribution."""
+
+from typing import Optional, Union
+
+import jax
+import jax.numpy as jnp
+from jaxtyping import Array, PRNGKeyArray
+
+from ..utils.math import mul_exp, multiply_no_nan, normalize
+from ._distribution import AbstractDistribution
+
+
+class Categorical(AbstractDistribution):
+    """Categorical distribution."""
```

Comment:
Could be good to add some description, i.e. so people know that this distribution describes a single categorical variable and such